### PR TITLE
Add Groups Support

### PR DESCRIPTION
**Pull request by neilkakkar**

### CODE REVIEW

**Review by macobo — CHANGES_REQUESTED** [review]
LGTM code-wise. 

I think we should make this optional given that group types has a low limit and is a paid feature though.

**Comment by neilkakkar** [comment]
I don't expect this to be run anywhere except in test instances, so the low limit of groups should be fine.
 
 About being paid, this wouldn't affect the UI at all? If you don't have the license, you wouldn't have access to groups.
 
 Just don't want to add options when defaults are good enough.

File: index.ts [hunk 1]
```diff
@@ -20,10 +20,14 @@ async function makeNoise() {
     const randomChoice = rnd(2)
     const userData = faker.helpers.contextualCard()
     
-    await posthog.capture('user signup', { ...userData, distinct_id: userData.username })
+    await posthog.capture('user signup', { ...userData, distinct_id: userData.username, $groups: {'organization': userData.company.name} })
```
Comment (macobo) [inline index.ts:23]: This needs to also groupidentify at some point - otherwise the group will have no properties. :)

**Comment by macobo** [comment]
It's a weird gotcha that might bite someone though - will leave it up to you, don't have real context on if this is used outside of local testing. Added one more note code-wise as well.

**Pull request closed without merging**